sql/plan/row_update_accumulator.go

```diff
@@ -209,6 +209,8 @@ func (a *accumulatorIter) Next() (sql.Row, error) {
 		row, err := a.iter.Next()
 		if err == io.EOF {
 			return sql.NewRow(a.updateRowHandler.okResult()), nil
+		} else if ErrInsertIgnore.Is(err) { // return err if its not an ignorable error
+			continue
 		}
```

Comment:
Move the `err != nil` up to make it an `else if` on this chain. Or you could move both of these inside of the `if err != nil`, returning the error if these two cases aren't matched.